agent-control/src/health/on_host/health_checker.rs

```diff
@@ -1,19 +1,22 @@
+use super::exec::ExecHealthChecker;
+use super::file::FileHealthChecker;
+use super::http::HttpHealthChecker;
 use crate::agent_type::runtime_config::health_config::{OnHostHealthCheck, OnHostHealthConfig};
+use crate::event::channel::EventConsumer;
 use crate::health::health_checker::{HealthChecker, HealthCheckerError};
 use crate::health::with_start_time::{HealthWithStartTime, StartTime};
 use crate::http::client::HttpClient;
 use std::path::PathBuf;
 
-use super::file::FileHealthChecker;
-use super::http::HttpHealthChecker;
-
 pub enum OnHostHealthChecker {
+    Exec(ExecHealthChecker),
     Http(HttpHealthChecker),
     File(FileHealthChecker),
 }
 
 impl OnHostHealthChecker {
     pub fn try_new(
+        exec_health_repository: EventConsumer<(String, HealthWithStartTime)>,
```

Comment:
Nit: We changed the "repository" approach to a "channel" approach, right? Should we change the name here?